packages/native/src/__tests__/useLinking.test.tsx

```diff
@@ -0,0 +1,77 @@
+import MockRouter from '../../../core/src/__tests__/__fixtures__/MockRouter';
+import Screen from '../../../core/src/Screen';
+import * as React from 'react';
+import {
+  NavigationContainer,
+  NavigationContainerRef,
+  useNavigationBuilder,
+} from '@react-navigation/core';
+import { render } from 'react-native-testing-library';
+import useLinking from '../useLinking';
+
+it('throws if multiple instances of useLinking are used', () => {
+  const TestNavigator = (props: any) => {
+    const { state, descriptors } = useNavigationBuilder(MockRouter, props);
+
+    return descriptors[state.routes[state.index].key].render();
+  };
+
+  const ref = React.createRef<NavigationContainerRef>();
+
+  const element = (
+    <NavigationContainer ref={ref}>
+      <TestNavigator initialRouteName="foo">
+        <Screen name="foo" component={() => null} />
+      </TestNavigator>
+    </NavigationContainer>
+  );
+
+  render(element);
+
+  const options = { prefixes: [] };
+
+  function Sample() {
+    useLinking(ref, options);
+    useLinking(ref, options);
+    return null;
+  }
+
+  const wrapper = <Sample />;
+
+  expect(() => render(wrapper)).toThrowError(
+    `Multiple instances of 'useLinking' are used`
+  );
+});
+
+it('not throws if single instance of useLinking are used', () => {
```

Comment:
Can't these 2 tests be only one, where double `useLinking` throws and single doesn't? The preparation looks the same.